File: SpringBoot-SecKill-Scala/src/main/scala/cn/edu/jxnu/seckill/controller/UserController.scala
package cn.edu.jxnu.seckill.controller

import org.springframework.web.bind.annotation.RestController
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.beans.factory.annotation.Autowired
import cn.edu.jxnu.seckill.redis.RedisService
import cn.edu.jxnu.seckill.service.SeckillUserService
import org.springframework.ui.Model
import cn.edu.jxnu.seckill.domain.SeckillUser
import cn.edu.jxnu.seckill.result.Result
import org.slf4j.LoggerFactory

/**
 * 用户控制器
 *
 * 压测专用
 * @author 梦境迷离.
 * @time 2018年5月21日
 * @version v1.0
 */
@RestController
@RequestMapping(Array("/user"))
class UserController @Autowired() (userService: SeckillUserService,
    redisService: RedisService) {

    private final val log = LoggerFactory.getLogger(classOf[UserController])

    /**
     * QPS:366.6 1000 * 10
     */
    @RequestMapping(Array("/info"))
    def info(model: Model, user: SeckillUser): Result[SeckillUser] = {

        log.info("用户user：" + user.toString())
        Result.success(user)
    }

}

File: seckill/src/main/java/cn/edu/jxnu/seckill/controller/UserController.java
package cn.edu.jxnu.seckill.controller;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;

import cn.edu.jxnu.seckill.domain.SeckillUser;
import cn.edu.jxnu.seckill.result.Result;
import io.swagger.annotations.Api;
import io.swagger.annotations.ApiOperation;

@Controller
@RequestMapping("/user")
@Api(value = "用户controller", tags = { "用户接口" })
public class UserController {

	@ApiOperation(value = "获取用户信息接口", notes = "获取用户信息")
	@RequestMapping("/info")
	@ResponseBody
	public Result<SeckillUser> doLogin(SeckillUser seckillUser) {
		return Result.success(seckillUser);
	}
}
